index.js

```diff
@@ -55,22 +84,59 @@ async function init() {
 
 	console.log('Source:', {repo, ref, dir});
 
+	if (!navigator.onLine) {
+		updateStatus('⚠ You are offline.');
+		throw new Error('User agent is offline');
+	}
+
 	updateStatus('Retrieving directory info…');
 
+	await ensureRepoIsAccessible(repo);
+
 	const files = await listContent.viaTreesApi(`${repo}#${ref}`, decodeURIComponent(dir), localStorage.token, ref);
 
-	updateStatus(`Downloading (0/${files.length}) files…`, '\n• ' + files.join('\n• '));
+	if (files.length > 0) {
+		updateStatus(`Downloading (0/${files.length}) files…`, '\n• ' + files.join('\n• '));
+	} else {
+		updateStatus('No files to download');
+		return;
+	}
 
 	let downloaded = 0;
-	const requests = await Promise.all(files.map(async path => {
-		const response = await fetch(`https://raw.githubusercontent.com/${repo}/${ref}/${path}`);
-		const blob = await response.blob();
+	let requests;
+	const controller = new AbortController();
+	try {
+		requests = await Promise.all(files.map(async path => {
+			const response = await fetch(
+				`https://raw.githubusercontent.com/${repo}/${ref}/${path}`,
+				{signal: controller.signal}
+			);
+
+			if (response.status >= 400) {
+				throw new FileDownloadError(path, response);
+			}
+
+			const blob = await response.blob();
+
+			downloaded++;
+			updateStatus(`Downloading (${downloaded}/${files.length}) files…`, path);
+
+			return {path, blob};
+		}));
+	} catch (error) {
+		controller.abort();
+
+		if (navigator.onLine === false) {
+			updateStatus('⚠ Could not download all files, network connection lost.');
+		} else if (error.name === 'FileDownloadError') {
```

Comment:
```suggestion
		} else if (error instanceof FileDownloadError) {
```

So you can drop the `name` prop as well.

Reply:
Yep, thought this as well. But basically all articles on "extend Errors correctly" say "set the name!!1!". 😁 Definitely not needed though, going to take this.

Reply:
Ok, you're right about the `name`

Reply:
Well then we can just as well keep the check as-is. 🤷‍♂

Reply:
`instanceof` is the right way to check that though.